src/java/org/apache/cassandra/service/snapshot/SnapshotManager.java

```diff
@@ -116,8 +108,15 @@ public synchronized void addSnapshot(TableSnapshot snapshot)
     @VisibleForTesting
     protected synchronized void loadSnapshots()
     {
-        logger.debug("Loading snapshots");
-        snapshotLoader.get().forEach(this::addSnapshot);
+        SnapshotLoader loader = new SnapshotLoader(DatabaseDescriptor.getAllDataFileLocations());
+        addSnapshots(loader.loadSnapshots());
+    }
+
+    @VisibleForTesting
+    protected synchronized void addSnapshots(Collection<TableSnapshot> snapshots)
+    {
+        logger.debug("Adding snapshots: {}", snapshots.stream().map(s -> s.getId()).collect(Collectors.toList()));
```

Comment:
To be explicit, this will be printed like: "Adding snapshots: ["snapshot1", "snapshot2","snapshot3"]" ?
I think there is some joining collector which just separates it by commas and leaves brackets. I consider it to be just nicer.

Reply:
Ehm ... should not it be the name of the snapshot rather than keyspace:table:id? I mean, we are adding _snapshots_, I do not see the _names_ of these snapshots or am I getting something wrong?